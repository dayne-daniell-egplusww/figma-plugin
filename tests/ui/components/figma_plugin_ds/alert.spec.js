import { shallowMount } from '@vue/test-utils';
import FigmaAlert from '~/ui/components/figma_plugin_ds/alert.vue';

describe('Figma alert component', () => {
  let wrapper;
  const findDismissButton = () => wrapper.find('.icon-button');

  function createComponent({ propsData = {} } = {}) {
    wrapper = shallowMount(FigmaAlert, {
      propsData,
    });
  }

  describe('renders', () => {
    it('default component', () => {
      createComponent();
      expect(wrapper.html()).toMatchSnapshot();
    });

    it.each`
      variant
      ${'danger'}
      ${'warning'}
      ${'info'}
      ${'success'}
    `('component with $variant variant class', ({ variant }) => {
      createComponent({ propsData: { variant } });

      expect(wrapper.classes()).toContain(`alert--${variant}`);
    });

    it('emits dismiss event when button clicked', () => {
      createComponent();
      findDismissButton().trigger('click');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.emitted().dismiss).toBeTruthy();
      });
    });
  });
});
