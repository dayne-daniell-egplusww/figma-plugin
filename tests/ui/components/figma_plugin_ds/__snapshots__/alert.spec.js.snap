// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figma alert component renders default component 1`] = `
"<div class=\\"d-flex align-items-center w-100 p-xxsmall pl-xsmall text-xsmall alert alert--info\\">
  <div class=\\"flex-grow-1\\"></div>
  <div class=\\"icon-button cursor-pointer\\">
    <div class=\\"icon icon--close cursor-pointer\\"></div>
  </div>
</div>"
`;
